src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php

```diff
@@ -432,6 +430,32 @@ protected function executeStatement(string $sql, array $parameters = [], array $
         return $stmt;
     }
 
+    private function executeInsert(string $sql, array $parameters = [], array $types = []): string
+    {
+        $this->driverConnection->beginTransaction();
+
+        try {
+            $this->driverConnection->executeStatement($sql, $parameters, $types);
+        } catch (TableNotFoundException $e) {
+            // create table
+            if ($this->autoSetup) {
+                $this->setup();
+            }
+
+            $this->driverConnection->executeStatement($sql, $parameters, $types);
```

Comment:
should we retry it when there is no auto-setup or should we rethrow the exception ?

Reply:
That is a good question. The existing code in `executeStatement()` that I copied didn't really make sense to me. I didn't want to modify that method, so I made `executeInsert()` and copied what was being done there to keep the behavior the exactly the same. If $this->autoSetup === false, then the retry will fail again, so I am not sure why it is retrying it when $this->autoSetup === false.

Reply:
note that executeStatement does not perform the autoSetup inside a transaction (because this is not safe cross-platform). if the goal is to reuse the same logic, maybe it should call the existing method.

Reply:
Do you think we should fix that in this PR too? It doesn't make sense the way the methods are written today. If `$this->autoSetup === false` then it should just throw the exception and not retry, and the retry should be moved inside of the `if ($this->autoSetup)`. Thoughts?

Reply:
I committed a change in fe1ed47da2cc742ade3dbbb666634d9af9a62f74 to see what it looks like.